From 3c9c47916c72628bc8b3f0e7f3e2e4b12062a42c Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Mon, 2 Jul 2012 22:50:40 +0300
Subject: [PATCH 2/3] require 'hoopl' with 'runWithFuel'

ghc-7.4.2 has 'hoopl-3.8.7.3' with accidentally dropped 'runWithFuel' form export.
Later versions have this export restored.

Patch fixes build failure on stock ghc-7.4.2:

> [ 8 of 45] Compiling HN.Optimizer.Dominator ( HN/Optimizer/Dominator.hs, dist/build/spl-test-hunit-exe/spl-test-hunit-exe-tmp/HN/Optimizer/Dominator.o )
> HN/Optimizer/Dominator.hs:15:48: Not in scope: `runWithFuel'

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 SPL.cabal |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/SPL.cabal b/SPL.cabal
index 7b01613..61d7d03 100644
--- a/SPL.cabal
+++ b/SPL.cabal
@@ -14,37 +14,37 @@ Executable spl-hnc
   Main-Is:         HNC.hs
   Other-Modules:   Bar
   Default-Language: Haskell2010
-  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.1, mtl == 2.*, parsec >= 3.1
+  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.4, mtl == 2.*, parsec >= 3.1
 
 Executable spl-test-hunit-exe
   Main-Is: Test/Main.hs
   Default-Language: Haskell2010
-  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.1, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
+  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.4, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
   Other-Modules:  Bar
 
 Test-Suite spl-test-hunit
   Main-Is: Test/Main.hs
   Type: exitcode-stdio-1.0
   Default-Language: Haskell2010
-  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.1, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
+  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.4, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
   Other-Modules:   Bar
 
 Executable	spli
   Main-Is:	SPLI.hs
   default-language: Haskell2010
-  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.1, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
+  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.4, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
   Other-Modules:   Bar
 
 Executable	splm
   Main-Is:	M.hs
   default-language: Haskell2010
-  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.1, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
+  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.4, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
   Other-Modules:   Bar
 
 Executable     spl-test-adhoc-hn
   Main-Is:        HN_Tests.hs
   default-language: Haskell98
-  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.1, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
+  Build-Depends:  base == 4.*, containers == 0.*, HUnit == 1.*, bytestring == 0.*, hoopl >= 3.8.7.4, mtl == 2.*, parsec >= 3.1, directory == 1.*, QuickCheck == 2.*
   Other-Modules:  Bar
 
 Executable	spl-test-adhoc-spl
-- 
1.7.8.6

